Gems/Atom/RHI/Vulkan/Code/Source/RHI/BindlessDescriptorPool.cpp

```diff
@@ -179,57 +176,55 @@ namespace AZ::Vulkan
             write.pImageInfo = &imageInfo;
             m_device->GetContext().UpdateDescriptorSets(m_device->GetNativeDevice(), 1, &write, 0, nullptr);
         }
-        return static_cast<uint32_t>(address.m_ptr);
+        return heapIndex;
     }
 
     uint32_t BindlessDescriptorPool::AttachReadBuffer(BufferView* view)
     {
         const uint32_t roBufferIndex = static_cast<uint32_t>(RHI::ShaderResourceGroupData::BindlessResourceType::ReadBuffer);
         AZStd::array<const RHI::ConstPtr<RHI::BufferView>, 1> span{ view };
-        RHI::VirtualAddress address;
+        uint32_t heapIndex = ~0u;
         {
             AZStd::lock_guard<AZStd::mutex> lock(m_mutex);
-            address = m_allocators[roBufferIndex].Allocate(1, 1).m_ptr;
-            AZ_Assert(address.IsValid(), "Bindless allocator ran out of space.");
+            heapIndex = static_cast<uint32_t>(m_allocators[roBufferIndex].Allocate(1, 1).m_ptr);
+            AZ_Assert(heapIndex != ~0u, "Bindless descriptor heap ran out of space.");

```

Comment:
I'm not a fan of this using internal knowledge about how a Virtual address works, I preferred the `IsValid` call